src/Payments.sol

```diff
@@ -510,6 +515,9 @@ contract Payments is
         rail.settledUpTo = block.number;
         rail.endEpoch = 0;
         rail.commissionRateBps = commissionRateBps;
+        rail.serviceFeeRecipient = serviceFeeRecipient == address(0)
```

Comment:
@SarveshLimaye Let's not do this. Let's revert `createRail ` if commission is non zero but `serviceFeeRecipient ` is 0 with a meaningful error message that the service contract dosen't end up doing something unexpected wrt collecting the comission.

Reply:
@aarshkshah1992 Sure makes sense. Implemented this in the recent commit